build.sbt

```diff
@@ -48,7 +48,19 @@ lazy val testLibs = Seq(
   "org.scalacheck" %% "scalacheck" % scalaCheckVersion % "test"
 )
 
+lazy val gemWarts =
+  Warts.allBut(
+    Wart.Any,                // false positives
+    Wart.Nothing,            // false positives
+    Wart.Product,            // false positives
+    Wart.Serializable,       // false positives
+    Wart.Recursion,          // false positives
```

Comment:
The problem is that monadic recursion (which is stacksafe via Free or whatever) can't be tail-recursive so it sets off this warning.